x-pack/legacy/plugins/ml/common/util/validators.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * Provides a validator function for maximum allowed input length.
+ * @param maxLength Maximum length allowed.
+ */
+export function maxLengthValidator(
+  maxLength: number
+): (value: string) => { maxLength: { requiredLength: number; actualLength: number } } | null {
+  return value => {
+    if (value && value.length > maxLength) {
+      return {
+        maxLength: {
+          requiredLength: maxLength,
+          actualLength: value.length,
+        },
+      };
+    } else {
```

Comment:
Absolute nitpick: You could just skip the `else` and return without it. For reference, here's the section on "return early from function" https://github.com/elastic/kibana/blob/master/style_guides/js_style_guide.md#returnthrow-early-from-functions